sdk/communication/communication-tiering/src/tieringClient.ts

```diff
@@ -0,0 +1,117 @@
+
+const isTieringClientOptions = (options: any): options is TieringClientOptions =>
+  options && !isKeyCredential(options) && !isTokenCredential(options);
+
+/**
+ * Client class for interacting with Azure Communication Services Tiering.
+ */
+export class TieringClient {
+  /**
+   * A reference to the auto-generated TieringClient HTTP client.
+   */
+  private readonly client: TieringGeneratedClient;
+
+  /**
+   * Initializes a new instance of the TieringClient class using a connection string.
+   *
+   * @param connectionString - Connection string to connect to an Azure Communication Service resource. (eg: endpoint=https://contoso.eastus.communications.azure.net/;accesskey=secret)
+   * @param options - Optional. Options to configure the HTTP pipeline.
+   */
+  public constructor(connectionString: string, options?: TieringClientOptions);
+
+  /**
+   * Initializes a new instance of the TieringClient class using an Azure KeyCredential.
+   *
+   * @param endpoint - The endpoint of the service (eg: https://contoso.eastus.communications.azure.net)
+   * @param credential - An object that is used to authenticate requests to the service. Use the Azure KeyCredential or `@azure/identity` to create a credential.
+   * @param options - Optional. Options to configure the HTTP pipeline.
+   */
+  public constructor(endpoint: string, credential: KeyCredential, options?: TieringClientOptions);
+
+  /**
+   * Initializes a new instance of the TieringClient class using an Azure KeyCredential.
+   *
+   * @param endpoint - The endpoint of the service (eg: https://contoso.eastus.communications.azure.net)
+   * @param credential - An object that is used to authenticate requests to the service. Use the Azure KeyCredential or `@azure/identity` to create a credential.
+   * @param options - Optional. Options to configure the HTTP pipeline.
+   */
+  public constructor(endpoint: string, credential: TokenCredential, options?: TieringClientOptions);
+
+  public constructor(
+    connectionStringOrUrl: string,
+    credentialOrOptions?: KeyCredential | TokenCredential | TieringClientOptions,
+    maybeOptions: TieringClientOptions = {}
+  ) {
+    const { url, credential } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
+    const options = isTieringClientOptions(credentialOrOptions)
+      ? credentialOrOptions
+      : maybeOptions;
+
+    const internalPipelineOptions: InternalClientPipelineOptions = {
+      ...options,
+      ...{
+        loggingOptions: {
+          logger: logger.info,
+        },
+      },
+    };
+
+    this.client = new TieringGeneratedClient(url, internalPipelineOptions);
+    const authPolicy = createCommunicationAuthPolicy(credential);
+    this.client.pipeline.addPolicy(authPolicy);
+  }
+
+  /**
+   * Get acquired numbers and limits for a resource.
+   *
+   * @param options - Additional request options.
+   */
+  public getAcquiredNumberLimits(
```

Comment:
This looks like a very thin wrapper around the generated code.  I wonder whether we should just create a Rest-Level-Client.

Reply:
How exactly would I do that?

Reply:
Oh, I see, if the client is lightweight its best to use RLC, one thing I should note is we do plan on growing this client in size as we go further down the line with Tiering. Would it make sense to keep it a RLC at that point? Because at that moment in time this SDK will be equal or greater than the recipient verification SDK in size.

Reply:
@classBabacar-msft it's not about lightweight or not, but whether you would do heavy customization on top of the generated code to provide better customer experiences.

Reply:
If we don't know if we will do heavy customization yet, wouldn't it be best to keep it like it is currently? Because I would prefer to keep it this way to match the other SDK's we created under "communication" in case we plan to make a big change to most of them in the future.

Reply:
that's a fair point. I am fine with having a High Level Client. We probably want a better  developer experiences for a HLC.